### ISSUE TITLE
daemon: Service registered in /etc/avahi/services does not seem to persist to NM hotspot after changing networks

### ISSUE THREAD

**Issue by smspillaz** [lathiat/avahi#183]
For our usecase we want to be able to create a wifi hotspot, advertise a service over dns-sd and have users of that hotspot be able to connect to it. However, if the service host (S) was already connected to another hotspot, existing services in /etc/avahi/services are not advertised on the new hotspot, even if the avahi daemon is restarted.

Steps to reproduce:

0. Have the service host (S) and clients (C) be connected to an existing wifi network
1. Create a service in /etc/avahi/services:

```
<service-group>

  <name>myservice</name>

  <service>
    <type>_servicetype._tcp</type>
    <port>1234</port>
  </service>

</service-group>
```

2. Confirm that the service is visible when running `avahi-browse -a` on C
3. Create a NetworkManager wifi hotspot on S (using gnome-control-center or similar)
4. Have C join S's hotspot.
5. After running `avahi-browse -a` on C the service is not visible C, even if the avahi-daemon is restarted via systemd or similar.

The only workaround appears to be configuring S so that it does not auto-connect to the existing wifi network the existing hotspot, the restarting NetworkManager on S, then setting up the hotspot on S per (3).

**Comment by smspillaz** [comment]
Corresponding NM bug: https://bugzilla.gnome.org/show_bug.cgi?id=796374

**Comment by evverx** [comment]
This issue is more than 5 years old so I'm not sure how to proceed here. I took a look at https://bugzilla.gnome.org/show_bug.cgi?id=796374 and it seems it was just automatically closed and I couldn't find any details. I'll close this issue but if it's still relevant could you open a new issue and attach avahi configs and debug logs on both machines? Pcaps showing mDNS packets sent/received by those machines would help too. My guess would be that the "MULTICAST" flag was missing on the interfaces or avahi could have been blocked by the firewall. It's hard to tell without logs/configs/pcaps.
